src/Data/Nat/Primality/Rough.agda

```diff
@@ -0,0 +1,57 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Natural numbers whose prime factors are all above a bound
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Data.Nat.Primality.Rough where
+
+open import Data.Nat.Base using (ℕ; suc; _≤_; _<_; z≤n; s≤s; _+_)
+open import Data.Nat.Divisibility using (_∣_; _∤_; ∣-trans; ∣1⇒≡1)
+open import Data.Nat.Induction using (<-rec; <-Rec)
+open import Data.Nat.Primality using (Prime; composite?)
+open import Data.Nat.Properties using (_≟_; <-trans; ≤∧≢⇒<; m<1+n⇒m<n∨m≡n)
+open import Data.Product.Base using (_,_)
+open import Data.Sum.Base using (inj₁; inj₂)
+open import Function.Base using (_∘_; flip)
+open import Relation.Nullary.Decidable.Core using (yes; no)
+open import Relation.Binary.PropositionalEquality.Core using (refl)
+
+private
+  variable
+    k m n : ℕ
+
+-- A number is k-rough if all its prime factors are greater than or equal to k
+_Rough_ : ℕ → ℕ → Set
+k Rough n = ∀ {d} → 1 < d → d < k → d ∤ n
+
+-- any number is 2-rough because all primes are greater than or equal to 2
+2-rough-n : ∀ n → 2 Rough n
+2-rough-n n {1} (s≤s ()) 1<2
+2-rough-n n {suc (suc d)} 1<d (s≤s (s≤s ()))
```

Comment:
Suggested refactoring, at the cost of a(nother, independently useful?) lemma, and a renaming:
```agda
¬0<n<1 : ∀ {n} → 0 < n → n < 1 → ⊥
¬0<n<1 z<s (s<s ())

2-rough : ∀ n → 2 Rough n
2-rough n (s<s 0<d) (s<s d<1) with () ← ¬0<n<1 0<d d<1
```

Reply:
I'd actually be surprised if there isn't a general form of that lemma in Data.Nat.Properties!

Reply:
I've made this change, and noticed that I could easily make the first argument implicit while I was at it